### DISCUSSION TITLE
Question: difference between Flax and Trax

### DISCUSSION THREAD

**Discussion by Borda** [google/flax#554] [General] [answered]
This is just a question as I cross over [Trax](https://github.com/google/trax) and [Flax](https://github.com/google/flax) today and both are google frameworks and both claim to be the deep learning framework in Jax. I noticed that the API is slightly different and Trax gives also an abstraction of the [training loop](https://github.com/google/trax#supervised-training)... Could you share some thoughts on the relation of these frameworks or eventual cooperation in a single user project?
 Thank you 🐰

**Comment by avital** [comment, marked as answer]
Hi @Borda - Trax and Flax are developed by different teams. There is no concrete plan right now to unify efforts. The APIs are quite different indeed, and one of the core developers of Flax (@levskaya) had previously worked on Trax. We explicitly designed Flax to be build of composable primitives rather than "inversion of control" type abstractions like training loops which are common in Trax.

Hope this helps!
